container-core/src/main/java/com/yahoo/metrics/simple/PointBuilder.java

```diff
@@ -27,18 +27,12 @@ public enum Discriminator {
     }
 
     PointBuilder(Point p) {
-        dimensions = new ArrayList<>();
-        location = new ArrayList<>();
         if (p != null) {
             int size = p.dimensionality();
             dimensions = new ArrayList<>(size+2);
```

Comment:
Consider rewriting as `new ArrayList<>(p.getDimensions())` if `size+2` matches length of `getDimensions()` output.

Reply:
That will never happen. p.getDimensions().size() == size == p.dimensionality().
I think intention is add room for 2 more than there were in the original Point.